milabench/executors.py

```diff
@@ -496,17 +506,33 @@ def __init__(
             **kwargs
     ) -> None:
         super().__init__(
-            None,
+            executors[0],
             **kwargs
         )
         self.executors = executors
 
     async def execute(self, **kwargs):
-        results = [] 
+        error_count = 0
+        def on_message(msg):
+            nonlocal error_count
+            
+            if msg.event == "error":
+                error_count += 1
+                
+            if msg.event == "end":
+                error_count += int(msg.data.get("return_code", 0))
+
+        loop = asyncio.get_running_loop()
+        loop._callbacks.append(on_message)
```

Comment:
Yeah I tried a few things but that ended up being the only thing that worked

Reply:
It'd require some modifications in alt_async, but one idea would be to have `await cmd(...)` throw an exception that has all of the errors in it. It would cause exceptions in other parts of the code that would have to catch them, though. Another idea would be to make it compatible with `async for ...`. Maybe in the future.